src/pubsub/mod_pubsub.erl

```diff
@@ -517,28 +514,30 @@ terminate_plugins(Host, ServerHost, Plugins, TreePlugin) ->
     gen_pubsub_nodetree:terminate(TreePlugin, Host, ServerHost),
     ok.
 
-send_loop(State) ->
-    receive
-        {send_last_pubsub_items, Recipient} ->
-            send_last_pubsub_items(Recipient, State),
-            send_loop(State);
-        {send_last_pep_items, Recipient, Pid} ->
-            send_last_pep_items(Recipient, Pid, State),
-            send_loop(State);
-        {send_last_items_from_owner, NodeOwner, Recipient} ->
-            send_last_items_from_owner(State#state.host, NodeOwner, Recipient),
-            send_loop(State);
-        stop ->
-            ok
-    end.
 
-send_last_pubsub_items(Recipient, #state{host = Host, plugins = Plugins})
-  when is_list(Plugins) ->
-    lists:foreach(
-      fun(PluginType) ->
-              send_last_pubsub_items_for_plugin(Host, PluginType, Recipient)
-      end,
-      Plugins).
+%% Currently HostType = ServerHost always (pubsub does not support dynamic domains yet)
+notify_worker(HostType, ServerHost, LUser, Request) ->
+    %% Signature: mongoose_wpool:cast(PoolType, HostType, Tag, HashKey, Request).
+    mongoose_wpool:cast(generic, HostType, pubsub_notify, LUser,
+                        {?MODULE, handle_msg, [HostType, ServerHost, Request]}).
+
+handle_msg(HostType, ServerHost, {send_last_pubsub_items, Recipient}) ->
+    %% Get subdomain
+    Host = host(HostType, ServerHost),
+    Plugins = gen_mod:get_module_opt(HostType, ?MODULE, plugins),
+    send_last_pubsub_items(Host, Recipient, Plugins);
+handle_msg(HostType, ServerHost, {send_last_pep_items, RecipientJID, RecipientPid}) ->
+    Host = host(HostType, ServerHost),
+    IgnorePepFromOffline = gen_mod:get_module_opt(HostType, ?MODULE, ignore_pep_from_offline),
+    send_last_pep_items(Host, IgnorePepFromOffline, RecipientJID, RecipientPid);
+handle_msg(HostType, ServerHost, {send_last_items_from_owner, NodeOwner, RecipientJID}) ->
+    Host = host(HostType, ServerHost),
+    send_last_items_from_owner(Host, NodeOwner, RecipientJID).
+
+send_last_pubsub_items(Host, Recipient, Plugins) ->
+    lists:foreach(fun(PluginType) ->
+          send_last_pubsub_items_for_plugin(Host, PluginType, Recipient)
+      end, Plugins).
```

Comment:
Minor: indentation is inconsistent here.

Reply:
good, I need to fix vim config on my new laptop finally :)